File: week11/javascript/bm4656/_26069.js
function solution(params) {
  let [n, ...arr] = params.toString().trim().split('\n');

  let set = new Set();

  set.add('ChongChong');

  for (let i = 0; i < n; i++) {
    const [a, b] = arr[i].split(' ');

    if (set.has(a)) {
      set.add(b);
    } else if (set.has(b)) {
      set.add(a);
    }
  }
  return set.size;
}

const case1 = `12
bnb2011 chansol
chansol chogahui05
chogahui05 jthis
jthis ChongChong
jthis jyheo98
jyheo98 lms0806
lms0806 pichulia
pichulia pjshwa
pjshwa r4pidstart
r4pidstart swoon
swoon tony9402
tony9402 bnb2011`;

console.log(solution(case1));

File: week11/java/src/Gawq/_26069.java
package Gawq;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.StringReader;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;

public class _26069 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new StringReader(src));
        Map<String, Integer> map = new HashMap<>();
        StringTokenizer st = null;
        int N = Integer.parseInt(br.readLine());
        int count = 1;
        String key = "ChongChong";
        map.put(key, count);

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String A = st.nextToken();
            String B = st.nextToken();

            if (!map.containsKey(A) && !map.containsKey(B)) {
                continue;
            }
            else {
                if (map.containsKey(A)) {
                    map.put(A, count++);
                    map.put(B, count);
                } else if (map.containsKey(B)) {
                    map.put(B, count++);
                    map.put(A, count);
                }
            }
        }
        System.out.println(map.size());
    }
    static String src = "12\n" +
            "bnb2011 chansol\n" +
            "chansol chogahui05\n" +
            "chogahui05 jthis\n" +
            "jthis ChongChong\n" +
            "jthis jyheo98\n" +
            "jyheo98 lms0806\n" +
            "lms0806 pichulia\n" +
            "pichulia pjshwa\n" +
            "pjshwa r4pidstart\n" +
            "r4pidstart swoon\n" +
            "swoon tony9402\n" +
            "tony9402 bnb2011";
}
